### PR TITLE
Server

### LINKED ISSUE #2: API
We should be able to run `polyglot --server` to start an api that returns the same json as in the cli version. 

**Request**
`http://localhost:3001/user=filhodanuvem&limit=2` 

**Response**
```
{
    "languages" : [
         "PHP"
         "Go"
     ]
}
```

### PR DESCRIPTION
**Pull request by santana125**
Related to Fixes #2 

I have added support to open an API server in order to use polyglot as a service.
The server was created using the [net/http](https://golang.org/pkg/net/http/) package. It's route can handle 2 arguments user (required) and limit (optional, defaults to 5).

You can change the server IP and PORT using the flags: --ip and --port, respectively.

image

This is my first real go code, I hope that it is good enough :smile:

### CODE REVIEW

**Review by filhodanuvem — CHANGES_REQUESTED** [review]
Nice work @santana125 ! 💪🏾

File: server/server.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	if len(repos) < 1 {
+		w.WriteHeader(http.StatusNotFound)
+		w.Write([]byte(`{"error": "This user has no public repositories"} `))
+		fmt.Printf("%v - %v - %v \n", req.Method, req.URL, http.StatusNotFound)
+		return
+	}
+
+	l := logrus.New()
+
+	stats := stats.GetStatisticsAsync("/tmp/polyglot", repos, l)
```
[the remark this answers is not in the document]
Comment (santana125) [reply server/server.go]: Oh sure, I forgot about that. Fixed

File: server/server.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte(`{"error": "` + err.Error() + `"} `))
+		fmt.Printf("%v - %v - %v \n", req.Method, req.URL, http.StatusInternalServerError)
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+	w.Write(responseJSON)
+
+	fmt.Printf("%v - %v - %v \n", req.Method, req.URL, http.StatusOK)
```
[the remark this answers is not in the document]
Comment (santana125) [reply server/server.go:86]: There is no reason. I will change then to logrus.

[the remark this answers is not in the document]
Comment (santana125) [reply server/server.go]: Done, I kept  fmt for the server start logo/text. Changed the requests to l.Printf and errors(including http.StatusInternalServerError) to l.Error

**Review by filhodanuvem — APPROVED** [review]

**Comment by filhodanuvem** [comment]
Thank you for the effort @santana125 🎉

**Pull request merged**